addons/hr_timesheet/tests/test_timesheet.py

```diff
@@ -438,3 +439,31 @@ def test_create_timesheet_with_archived_employee(self):
 
         with self.assertRaises(UserError):
             timesheet.employee_id = self.empl_employee2
+
+    def test_remaining_hours_on_project(self):
+        ''' Test the remaining hours on the project '''
+
+        self.task1.planned_hours = 4.0
+
+        self.assertEqual(self.project_customer.remaining_hours, 4.0, 'Remaining hours should be 4.0 before adding timesheet')
+
+        self.env['account.analytic.line'].create({
+            'project_id': self.project_customer.id,
+            'task_id': self.task1.id,
+            'name': 'test',
+            'unit_amount': 2,
+        })
+        self.assertEqual(self.project_customer.remaining_hours, 2.0, 'Remaining hours should be 2.0 after adding timesheet')
+
+        task_child = self.env['project.task'].create({
+            'name': 'Task Child',
+            'project_id': self.project_customer.id,
+            'parent_id': self.task1.id
+        })
+        self.env['account.analytic.line'].create({
+            'project_id': self.project_customer.id,
+            'task_id': task_child.id,
+            'name': 'test',
+            'unit_amount': 2,
+        })
+        self.assertEqual(self.project_customer.remaining_hours, 2.0, 'Remaining hours should be 2.0 after adding timesheet')
```

Comment:
Why is it not 0? 🤔

Reply:
Because, as I understood, the remaining hours in the parent task should include the remaining hours of a sub task.
And so, at the end, we have 2.0 remaining (even if 2.0 on the timesheet of the parent AND child task)

Reply:
I am not sure to understand your explanation. Is it not because the planned_hours is equal to 0 in the subtask and so we do not take into account the timesheet linked to that subtask? 🤔

Reply:
I updated the test and how we calculate the reamining_hours (like discussed in MP):wink:

Reply:
Wonderful, thanks again. 🙂